Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -796,11 +813,107 @@ namespace xr
             }
         }
 
+        void UpdatePlanes(std::map<NativePlaneIdentifier, Plane*>& existingPlanes, std::vector<Plane>& newPlanes, std::vector<Plane*>& deletedPlanes)
+        {
+            if (!IsTracking())
+            {
+                return;
+            }
+
+            // First check if any existing planes have been subsumed by another plane, if so add them to the list of deleted planes
+            CheckForSubsumedPlanes(existingPlanes, deletedPlanes);
+
+            // Next check for updated planes, and update their pose and polygon or create a new plane if it does not yet exist.
+            ArFrame_getUpdatedTrackables(session, frame, AR_TRACKABLE_PLANE, trackableList);            
+            int32_t size{};
+            ArTrackableList_getSize(session, trackableList, &size);
+            for (int i = 0; i < size; i++)
+            {
+                // Get the plane.
+                ArTrackable* trackable;
+                ArTrackableList_acquireItem(session, trackableList, i, &trackable);
+                auto planeTrackable = reinterpret_cast<ArPlane*>(trackable);
```

Comment:
I would put `ArTrackable* trackable` in its own scope so we don't have two variables referring to the same object. This will help force the following code to be a bit easier to follow.
```suggestion
                ArPlane* planeTrackable{};
                {
                    ArTrackable* trackable{};
                    ArTrackableList_acquireItem(session, trackableList, i, &trackable);
                    planeTrackable = reinterpret_cast<ArPlane*>(trackable);
                }
```